packages/generator/src/vdm-types.ts

```diff
@@ -131,7 +131,8 @@ export enum VdmReturnTypeCategory {
   ENTITY,
   COMPLEX_TYPE,
   EDM_TYPE,
-  VOID
+  VOID,
+  ENTITY_NOT_DESERIALIZABLE
```

Comment:
`ENTITY_NOT_DESERIALIZABLE` does not really sound like a category to me... See the overall comment from my review.